terragrunt/modules/docs-rs/web-server.tf

```diff
@@ -1,88 +1,92 @@
-#     DOCSRS_STATIC_CLOUDFRONT_ID = "${aws_cloudfront_distribution.static.id}"
-#   }
+  environment = {
+    DOCSRS_PREFIX               = "/tmp"
+    DOCSRS_STORAGE_BACKEND      = "s3"
+    DOCSRS_LOG                  = "docs_rs=debug,rustwide=info"
+    RUST_BACKTRACE              = "1"
+    DOCSRS_STATIC_CLOUDFRONT_ID = "${aws_cloudfront_distribution.static.id}"
+  }
 
-#   secrets = {
-#     DOCSRS_GITHUB_ACCESSTOKEN = "/prod/docs-rs/github-access-token"
-#   }
+  secrets = {
+    DOCSRS_GITHUB_ACCESSTOKEN = "/prod/docs-rs/github-access-token"
+  }
 
-#   computed_secrets = {
-#     DOCSRS_DATABASE_URL = aws_ssm_parameter.connection_url.arn
-#   }
+  computed_secrets = {
+    DOCSRS_DATABASE_URL = aws_ssm_parameter.connection_url.arn
+  }
 
-#   expose_http = {
-#     container_port = 80
-#     prometheus     = "/about/metrics"
-#     domains        = ["docs-rs-web-prod.infra.rust-lang.org"]
+  expose_http = {
+    container_port = 80
+    prometheus     = "/about/metrics"
+    domains        = [local.web_domain]
+    zone_id        = var.zone_id
 
-#     health_check_path     = "/"
-#     health_check_interval = 5
-#     health_check_timeout  = 2
-#   }
+    health_check_path     = "/"
+    health_check_interval = 5
+    health_check_timeout  = 2
+  }
 
-#   # Allow database access
-#   additional_security_group_ids = [aws_security_group.web.id]
-# }
+  # Allow database access
+  additional_security_group_ids = [aws_security_group.web.id]
+}
 
-# resource "aws_iam_role_policy" "web" {
-#   role = module.web.role_id
-#   name = "inline"
+resource "aws_iam_role_policy" "web" {
+  role = module.web.role_id
+  name = "inline"
 
-#   policy = jsonencode({
-#     Version = "2012-10-17"
-#     Statement = [
-#       {
-#         Effect = "Allow"
-#         Action = [
-#           "s3:PutObject",
-#           "s3:GetObjectTagging",
-#           "s3:PutObjectTagging",
-#           "s3:GetObject",
-#           "s3:PutObjectAcl",
-#           "s3:ListBucket",
-#           "s3:DeleteObject",
-#         ]
-#         Resource = [
-#           "arn:aws:s3:::rust-docs-rs",
-#           "arn:aws:s3:::rust-docs-rs/*",
-#         ]
-#       },
-#       {
-#         Effect = "Allow"
-#         Action = [
-#           "cloudfront:CreateInvalidation",
-#           "cloudfront:GetInvalidation",
-#           "cloudfront:ListInvalidations",
-#         ]
-#         Resource = [
-#           aws_cloudfront_distribution.webapp.arn,
-#           aws_cloudfront_distribution.static.arn,
-#         ]
-#       },
-#     ]
-#   })
-# }
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "s3:PutObject",
+          "s3:GetObjectTagging",
+          "s3:PutObjectTagging",
+          "s3:GetObject",
+          "s3:PutObjectAcl",
+          "s3:ListBucket",
+          "s3:DeleteObject",
+        ]
+        Resource = [
+          "arn:aws:s3:::rust-docs-rs",
```

Comment:
Are we sharing the S3 bucket between staging and production? This feels like something that should come from either a variable or a resource.

Reply:
I believe these will be in separate accounts.

Reply:
Bucket names are globally unique. So it doesn't matter in what account this is, it would still refer to the one `rust-docs-rs` bucket. Not the first time we're running into this. 😞